src/main/java/org/opensearch/agent/tools/PPLTool.java

```diff
@@ -213,7 +222,17 @@ public <T> void run(Map<String, String> parameters, ActionListener<T> listener)
             ));
         }, e -> {
             log.info("fail to get mapping: " + e);
-            listener.onFailure(e);
+            String error_message = e.getMessage();
+            if (error_message.contains("no such index")) {
+                listener
+                    .onFailure(
+                        new IllegalArgumentException(
+                            "Return this final answer to human directly and do not use other tools: 'Please provide index name'. Please try to directly send this message to human to ask for index name"
```

Comment:
Should we change the message to something like: `the index is not exist, please provide a correct index name ...`